### PR TITLE
[RHICOMPL-832] Improve validation for "Create Policy"

### PR DESCRIPTION
**Pull request by bastilian**
* Adds validation for rules to be selected
 * Fixes compliance threshold validation

### CODE REVIEW

File: src/SmartComponents/CreatePolicy/CreatePolicy.js [hunk 1]
```diff
@@ -99,9 +102,10 @@ const selector = formValueSelector('policyForm');
 export default connect(
     state => ({
         benchmark: selector(state, 'benchmark'),
-        profile: selector(state, 'profile'),
+        complianceThreshold: selector(state, 'complianceThreshold'),
         name: selector(state, 'name'),
+        profile: selector(state, 'profile'),
         refId: selector(state, 'refId'),
-        selectedRuleRefIds: selector(state, 'selectedRuleRefIds') || []
```
Comment (bastilian) [inline src/SmartComponents/CreatePolicy/CreatePolicy.js:105]: This changes makes the rules validate-able.

File: src/SmartComponents/CreatePolicy/validate.js [hunk 2]
```diff
@@ -6,13 +6,9 @@ export const validateFirstPage = (benchmark, profile) => {
...
 
-export const validateSecondPage = (name, refId) => {
-    if (!name || !refId) {
-        return false;
-    } else {
-        return true;
-    }
-};
+export const validateSecondPage = (name, refId, complianceThreshold) => (
+    !name || !refId  || !complianceThreshold || complianceThreshold > 100 || complianceThreshold < 0 ? false : true
```
Comment (bastilian) [inline src/SmartComponents/CreatePolicy/validate.js:10]: This is the validation that will prevent moving forward in the wizard with an invalid threshold.

**Comment by bastilian** [comment]
@akofink Thank you testing and reviewing! I had committed part of the changes to #685 🤦 

It now gives feedback when the threshold is invalid and allows to move forward if it's valid:

Kapture 2020-08-27 at 16 31 33

The other cases where the second screen would not allow to move forward is if name or refId aren't set due to not loading properly. This is the same behaviour we have now in master.

We should definitely extend/improve validations here and in the edit policy modal, but that would be much easier with/after moving to something, probably most likely, data driven forms, which also integrates with the PF4 Wizard.

**Comment by bastilian** [comment]
Just a side note: The hashed filename configuration is causing very odd issues. 
When testing be sure to clear any caches and best apply #719 till it is merged.

File: src/SmartComponents/ProfileThresholdField/ProfileThresholdField.js [hunk 3]
```diff
@@ -7,22 +7,18 @@ import propTypes from 'prop-types';
...
-            this.setState({ validThreshold: true, threshold: round(threshold, 1) });
-        }
+    state = {
+        validThreshold: (this.props.previousThreshold < 101 && this.props.previousThreshold > 0),
+        threshold: round(this.props.previousThreshold || 100, 1)
     };
 
+    handleThresholdChange = (threshold) => (
+        this.setState({
+            validThreshold: (threshold < 101 && threshold > 0),
```
Comment (akofink) [inline src/SmartComponents/ProfileThresholdField/ProfileThresholdField.js:17]:
```suggestion
            validThreshold: (threshold < 101 && threshold >= 0),
```

Comment (bastilian) [reply src/SmartComponents/ProfileThresholdField/ProfileThresholdField.js:17]: 👍 will update that and actually extract it to use the same everywhere this occurs.

Comment (bastilian) [reply src/SmartComponents/ProfileThresholdField/ProfileThresholdField.js]: Update! thanks for the suggestion!

**Pull request merged**